pkg/jmxfetch/jmxfetch.go

```diff
@@ -206,7 +202,16 @@ func (j *JMXFetch) Start(manage bool) error {
 	case ReporterJSON:
 		reporter = "json"
 	default:
-		reporter = fmt.Sprintf("statsd:%s:%s", bindHost, config.Datadog.GetString("dogstatsd_port"))
+		socketPath := config.Datadog.GetString("dogstatsd_socket")
```

Comment:
Now that I think about it more, I think it would be better to check that the UDS dogstatsd listener was indeed successfully initialized before configuring JMXFetch to report to it, since when `dogstatsd_socket` is configured there are a number of additional cases that can make the UDS listener fail to start (see https://github.com/DataDog/datadog-agent/blob/8493176ff4b8c2ec531e28fbe2e446c31daba3a1/pkg/dogstatsd/listeners/uds_common.go#L58-L87). And make JMXFetch report with UDP otherwise.

Otherwise, for a user with a non-working `dogstatsd_socket` setting, this change would break JMXFetch reporting.

In terms of implementation: maybe the `dogstatsd` package could have a new sub-package (named `features`?) that provides an exported function exposing the status of the UDS listener (unknown/ok/not running), and the jmxfetch logic here would poll this function for a few seconds until it returns either `ok` or `not running`? WDYT?

Reply:
I think this makes sense and seems reasonable. Will make the change.